py3dtilers/Common/feature.py

```diff
@@ -229,13 +229,47 @@ def add_materials(self, materials):
         """
         self.materials.extend(materials)
 
+    def add_material(self, material):
+        """
+        Extend the materials of this object with a GltF material.
+        :param material: a GlTFMaterial
+        """
+        self.materials.append(material)
+
     def get_material(self, index):
         """
         Get the material at the index.
         :param index: the index (int) of the material
+        :return: a glTF material
         """
         return self.materials[index]
 
+    def is_material_registered(self, material):
+        """
+        Check if a material is already set in materials array
+        :param material: a GlTFMaterial
+        :return: bool
+        """
+        for mat in self.materials:
+            if(mat.rgba == material.rgba).all():
```

Comment:
We could different materials having the same RGBA values (different roughness/metallic factors, different textures...). Maybe we could override the `==` operator of `GltfMaterial`, where we would test if all the attributes of 2 materials are different

Reply:
Yes, I totally agree. It was the first lazy attempt to avoid storing the same material.